addons/point_of_sale/models/pos_config.py

```diff
@@ -424,8 +425,30 @@ def write(self, vals):
                     if key == 'use_pricelist' and vals[key]:
                         continue
                     if key == 'available_pricelist_ids':
-                        removed_pricelist = set(self.available_pricelist_ids.ids) - set(vals[key][0][2])
-                        if len(removed_pricelist) == 0:
+                        will_unlink_a_pricelist = \
+                            (
+                                (not isinstance(vals[key], list) or len(vals[key]) == 0)
+                                and self.available_pricelist_ids
```

Comment:
```suggestion
                                not isinstance(vals[key], list) and (not vals[key] and self.available_pricelist_ids)
```

if `vals[key]` is not a list, then it should be `False`. So if it's `False` and there is `available_pricelist_ids`, it will be unlinked. Can you confirm?

Reply:
If vals[key] is not a list, then available_pricelist_ids will be cleared or altered in a currently unknown way (False, None or any illegal value). Therefore, will_unlink_a_pricelist should be True to support current and potential future possible values (evolution of Odoo ORM) of available_pricelist_ids.
Added to that, with your suggestion, will_unlink_a_pricelist would be False for an empty list, which is wrong.